### DISCUSSION TITLE
Installation Issues on Windows: Vue Package Version Mismatch and OpenSSL Error (solved)

### DISCUSSION THREAD

**Discussion by 3dkrender** [worldwide-asset-exchange/docs#137] [General]
### Issues with Repository Installation (on Windows)

**Runtime Environment:**

Windows 11
Node V.18.15.0

**Problem Description**

I have encountered some issues while trying to install the repository.

When executing `npm install`, the result was:

```
55 vulnerabilities (1 low, 31 moderate, 20 high, 3 critical)
```

And when attempting to run `npm run docs:dev`, the following error appeared:

```
Error:

Vue packages version mismatch:

- vue@3.3.4
- vue-server-renderer@2.7.14
```
 --- 

To solve it, I chose to use Yarn instead of npm:

```
yarn install
yarn upgrade
```

Now, when running the server, this error appeared:

```
error:0308010C:digital envelope routines::unsupported
```

Solution: Enable legacy OpenSSL provider.

On Unix-like systems (Linux, macOS, Git bash, etc.):

```
export NODE_OPTIONS=--openssl-legacy-provider
```

On Windows command prompt:
```
set NODE_OPTIONS=--openssl-legacy-provider
```

On PowerShell:
```
$env:NODE_OPTIONS = "--openssl-legacy-provider"
```

--- 

Now everything is OK. I hope this information can be helpful.

**Comment by rakeden** [comment]
hello @3dkrender and thanks for the issue! 
I'll leave it open, as other devs might encounter it until fixed.

The vulns reported during `npm install` are indeed correct, not solvable right now though.

For the other reported error it is recommended to run the project with the specified node engine inside package.json
```
"engines": {
  "node": ">=16.19.0"
},
```
I recommend using [nvm](https://github.com/nvm-sh/nvm) to manage local node engines.

**or alternatively:**

As per your suggestion - Add the flag to the dev and build command:
`NODE_OPTIONS=--openssl-legacy-provider`

If needed, CI errors can be suppressed when installing deps by using:
`CI=false npm install`
